ray-operator/controllers/ray/raycluster_controller_fake_test.go

```diff
@@ -365,6 +373,149 @@ func tearDown(t *testing.T) {
 	PrioritizeWorkersToDelete = false
 }
 
+func TestReconcile_UnhealthyEvent(t *testing.T) {
+	setupTest(t)
+	defer tearDown(t)
+
+	testPodName := "eventPod"
+	testPodEventName := fmt.Sprintf("%s.15f0c0c5c5c5c5c5", testPodName)
+
+	// add a pod in a different namespace
+	newPods := []runtime.Object{
```

Comment:
The specifications of the Pods are very similar (some parts are the same.). Can we use variables instead of expanding all of them?

Reply:
This is a test case. In this test case, we have `common.RayNodeLabelKey: "yes",` specified. I know some parts are the same but putting them in a separate place might not be good for test cases.

Reply:
Let me explain it more. My meaning is that:

```go
spec :=  corev1.PodSpec{
  Containers: []corev1.Container{
	  {
		  Name:    "ray-worker",
		  Image:   "rayproject/ray:2.2.0",
		  Command: []string{"echo"},
		  Args:    []string{"Hello Ray"},
	  },
  },
}

&corev1.Pod {
   ...
   Spec: spec (Maybe need to use Deep copy)
}

&corev1.Pod {
   ...
   Spec: spec (Maybe need to use Deep copy)
}

```

instead of 

```go
&corev1.Pod {
   ...
   Spec: corev1.PodSpec{
    Containers: []corev1.Container{
	    {
		    Name:    "ray-worker",
		    Image:   "rayproject/ray:2.2.0",
		    Command: []string{"echo"},
		    Args:    []string{"Hello Ray"},
	    },
    },
  }
}

&corev1.Pod {
   ...
   Spec: corev1.PodSpec{
    Containers: []corev1.Container{
	    {
		    Name:    "ray-worker",
		    Image:   "rayproject/ray:2.2.0",
		    Command: []string{"echo"},
		    Args:    []string{"Hello Ray"},
	    },
    },
  }
}

```

Reply:
There are a lot of technical debts in KubeRay now. It's good for us to reduce the system's complexity in every PR : )